src/wp-includes/class-wp-theme-json.php

```diff
@@ -930,14 +939,22 @@ protected static function do_opt_in_into_settings( &$context ) {
 	 * @since 5.9.0 Added the `$valid_block_names` and `$valid_element_name` parameters.
 	 * @since 6.3.0 Added the `$valid_variations` parameter.
 	 * @since 6.6.0 Updated schema to allow extended block style variations.
+	 * @since 6.9.0 Added caching to improve performance.
 	 *
 	 * @param array $input               Structure to sanitize.
 	 * @param array $valid_block_names   List of valid block names.
 	 * @param array $valid_element_names List of valid element names.
 	 * @param array $valid_variations    List of valid variations per block.
+	 *
 	 * @return array The sanitized output.
 	 */
 	protected static function sanitize( $input, $valid_block_names, $valid_element_names, $valid_variations ) {
+
+		$input_cache_key = md5( wp_json_encode( $input ) );
```

Comment:
I believe the other args should be included in the cache key, should they not?

```suggestion
		$input_cache_key = md5( wp_json_encode( compact( 'input', 'valid_block_names', 'valid_element_names', 'valid_variations' ) ) );
```